elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/filter/HQLFilterOperation.java

```diff
@@ -60,4 +65,47 @@ public String applyAll(Set<Predicate> predicates) {
 
         return filterString.toString();
     }
+
+    public String apply(FilterExpression filterExpression) {
+        HQLQueryVisitor visitor = new HQLQueryVisitor();
+        return "WHERE " + filterExpression.accept(visitor);
+
+    }
+
+    /**
+     * Filter expression visitor which builds an HQL query.
+     */
+    public class HQLQueryVisitor implements Visitor<String> {
```

Comment:
THis visitor calls apply on the HQLFilterOperation - so it cannot be static as it stands.